integration/query_test.go

```diff
@@ -67,3 +68,206 @@ func TestQueryCount(t *testing.T) {
+			},
+		},
+		"DoubleWhole": {
+			command: bson.D{
+				{"find", 42.0},
+				{"projection", bson.D{{"value", "some"}}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "collection name has invalid type double",
+			},
+		},
+		"Binary": {
+			command: bson.D{
+				{"find", primitive.Binary{}},
+				{"projection", bson.D{{"value", "some"}}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "collection name has invalid type binData",
+			},
+		},
+		"ObjectID": {
+			command: bson.D{
+				{"find", primitive.ObjectID{}},
+				{"projection", bson.D{{"value", "some"}}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "collection name has invalid type objectId",
+			},
+		},
+		"Bool": {
+			command: bson.D{
+				{"find", true},
+				{"projection", bson.D{{"value", "some"}}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "collection name has invalid type bool",
+			},
+		},
+		"Date": {
+			command: bson.D{
+				{"find", time.Now()},
+				{"projection", bson.D{{"value", "some"}}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "collection name has invalid type date",
+			},
+		},
+		"Null": {
+			command: bson.D{
+				{"find", nil},
+				{"projection", bson.D{{"value", "some"}}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "collection name has invalid type null",
+			},
+		},
+		"Regex": {
+			command: bson.D{
+				{"find", primitive.Regex{Pattern: "/foo/"}},
+				{"projection", bson.D{{"value", "some"}}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "collection name has invalid type regex",
+			},
+		},
+		"Int": {
+			command: bson.D{
+				{"find", int32(42)},
+				{"projection", bson.D{{"value", "some"}}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "collection name has invalid type int",
+			},
+		},
+		"Timestamp": {
+			command: bson.D{
+				{"find", primitive.Timestamp{}},
+				{"projection", bson.D{{"value", "some"}}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "collection name has invalid type timestamp",
+			},
+		},
+		"Long": {
+			command: bson.D{
+				{"find", int64(42)},
+				{"projection", bson.D{{"value", "some"}}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "collection name has invalid type long",
+			},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var actual bson.D
+			err := collection.Database().RunCommand(ctx, tc.command).Decode(&actual)
+			if tc.err != nil {
+				AssertEqualError(t, *tc.err, err)
+				return
+			}
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestQuerySort(t *testing.T) {
```

Comment:
```suggestion
func TestQueryBadSortType(t *testing.T) {
```

Since we will have a separate test for sorting that could be named `TestQuerySort`